data-prepper-plugins/opensearch/src/test/java/com/amazon/dataprepper/plugins/sink/opensearch/OpenSearchIntegrationHelper.java

```diff
@@ -0,0 +1,254 @@
+        final Settings settings = Settings.Builder.EMPTY_SETTINGS;
+        return buildClient(settings, httpHosts);
+    }
+
+    private static RestClient buildClient(final Settings settings, final HttpHost[] hosts) throws IOException {
+        final RestClientBuilder builder = RestClient.builder(hosts);
+        if (isOSBundle()) {
+            configureHttpsClient(builder, settings);
+        } else {
+            configureClient(builder, settings);
+        }
+
+        builder.setStrictDeprecationMode(true);
+        return builder.build();
+    }
+
+    static boolean isOSBundle() {
+        final boolean osFlag = Optional.ofNullable(System.getProperty("os"))
+                .map("true"::equalsIgnoreCase).orElse(false);
+        if (osFlag) {
+            // currently only external cluster is supported for security enabled testing
+            if (!Optional.ofNullable(System.getProperty("tests.rest.cluster")).isPresent()) {
+                throw new RuntimeException("cluster url should be provided for security enabled OpenSearch testing");
+            }
+        }
+
+        return osFlag;
+    }
+
+    private static String getProtocol() {
+        return isOSBundle() ? "https" : "http";
+    }
+
+
+    private static void configureHttpsClient(final RestClientBuilder builder, final Settings settings) throws IOException {
+        final Map<String, String> headers = ThreadContext.buildDefaultHeaders(settings);
+        final Header[] defaultHeaders = new Header[headers.size()];
+        int i = 0;
+        for (Map.Entry<String, String> entry : headers.entrySet()) {
+            defaultHeaders[i++] = new BasicHeader(entry.getKey(), entry.getValue());
+        }
+        builder.setDefaultHeaders(defaultHeaders);
+        builder.setHttpClientConfigCallback(httpClientBuilder -> {
+            final String userName = Optional
+                    .ofNullable(System.getProperty("user"))
+                    .orElseThrow(() -> new RuntimeException("user name is missing"));
+            final String password = Optional
+                    .ofNullable(System.getProperty("password"))
+                    .orElseThrow(() -> new RuntimeException("password is missing"));
+            final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+            credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(userName, password));
+            try {
+                return httpClientBuilder
+                        .setDefaultCredentialsProvider(credentialsProvider)
+                        // disable the certificate since our testing cluster just uses the default security configuration
+                        .setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE)
+                        .setSSLContext(SSLContextBuilder.create().loadTrustMaterial(null, (chains, authType) -> true).build());
+            } catch (Exception e) {
+                throw new RuntimeException(e);
```

Comment:
Why catch an exception only to throw a generic runtime exception?

Reply:
This was already part of `OpenSearchSinkIT` and I moved this method. There is little value here since these are all tests.